framework/decode/screenshot_handler.cpp

```diff
@@ -38,10 +39,41 @@ static constexpr size_t kUnormIndex = 0;
 static constexpr size_t kSrgbIndex  = 1;
 
 const VkFormat kImageFormats[][2] = {
-    // Vulkan image formats for ScreenshotFormat::kBmp
-    { VK_FORMAT_B8G8R8A8_UNORM, VK_FORMAT_B8G8R8A8_SRGB }
+    // Vulkan image formats for util::ScreenshotFormat::kBmp
+    { VK_FORMAT_B8G8R8A8_UNORM, VK_FORMAT_B8G8R8A8_SRGB },
+    // Vulkan image formats for util::ScreenshotFormat::kPng
+    { VK_FORMAT_R8G8B8A8_UNORM, VK_FORMAT_R8G8B8A8_SRGB }
 };
 
+inline void WriteImageFile(const std::string&     filename,
+                           util::ScreenshotFormat file_format,
+                           uint32_t               width,
+                           uint32_t               height,
+                           uint64_t               size,
+                           void*                  data)
+{
+    switch (file_format)
+    {
+        default:
```

Comment:
Would it not be better to have no default and let this be a compile-time warning (promoted to an error by warnings as errors)? Same with the DX12 case in another file. Sorry not to mention earlier, I just noticed.

I am not blocking on this, as interested in learning from contrary opinions as in asserting my own here!

Reply:
`kPng` in `framework/util/options.h` would also need to be wrapped with `GFXRECON_ENABLE_PNG_SCREENSHOT` or this code would error without `default`

Reply:
Presumably kPng could never reach this function if the code which emits it and is wrapped in `GFXRECON_ENABLE_PNG_SCREENSHOT` is not compiled-in to a build. But I'm either/or on this and I see where you are coming from.